server/serverHelpers.js

```diff
@@ -1,38 +1,74 @@
-const getReviews = (product) => {
-  // get all reviews, user info, images, comments for product
-};
+import Promise from 'bluebird';
 
-const getAggregate = (product) => {
-  // get aggregate score & review quantity
-};
+import { db } from '../database/connection';
+
+const getAggregate = product => new Promise((resolve) => {
+  db.query(`SELECT score, qty FROM aggregates WHERE product_id=${product};`, (err, data) => {
+    if (err) return 404;
+    resolve(data);
+  });
+});
+
+const getReviews = (product, callback) => new Promise((resolve) => {
+  db.query(`SELECT * FROM reviews INNER JOIN users ON reviews.user_id=users.id WHERE product_id=${product}`, (err, data) => {
+    if (err) return 404;
+    resolve(callback(data));
+  });
+});
 
-const updateAggregate = (product, review) => {
-  // TODO: write record @ product id, filtering & averaging reviews by product id
+const getImages = (reviews) => {
+  const results = {};
+  results.reviews = reviews;
+  let queryString = '';
+  return new Promise((resolve) => {
+    reviews.forEach((review) => {
+      queryString += `SELECT review_id, title, url FROM images WHERE review_id=${review.id};`;
```

Comment:
Possible to refactor the querystring on the outside to include 'SELECT review_id, title, url FROM images WHERE review_id='  and just append the review.id for each review.  Might make the code a little more clear!

Reply:
Will come back to this if time permits. Since it's a multi-line query, refactoring would be more than a one-liner